src/lib.rs

```diff
@@ -300,41 +330,95 @@ impl Analyzer {
             Using(using) => usings.push((*using.clone(), parent)),
             StraySemicolon(_loc) => todo!(),
             PragmaDirective(_, _, _) => {}
-            ImportDirective(import) => imported.extend(self.parse_import(import)),
+            ImportDirective(import) => imported.extend(self.parse_import(import, &current_path)),
         }
         (sup_node, func_nodes, usings)
     }
 
     pub fn parse_import(
         &mut self,
         import: &Import,
+        current_path: &PathBuf,
     ) -> Vec<(Option<NodeIdx>, String, String, usize)> {
         match import {
-            Import::Plain(path, _) => {
-                // println!("path: {:?}", path);
-                let sol = fs::read_to_string(path.string.clone()).unwrap_or_else(|_| {
-                    panic!("Could not find file for dependency: {:?}", path.string)
+            Import::Plain(import_path, _) => {
+                // println!("import_path: {:?}", import_path);
+
+                let remapping = self
+                    .remappings
+                    .clone()
+                    .into_iter()
+                    .find(|x| import_path.string.starts_with(&x.name));
+
+                let remapped = if remapping.is_some() {
+                    let unwrapped = remapping.clone().unwrap();
+                    Remapping::into_relative(remapping.unwrap(), &self.root)
+                        .path
+                        .relative()
+                        .join(&import_path.string.replacen(&unwrapped.name.clone(), "", 1))
+                } else {
+                    PathBuf::from(current_path.clone())
+                        .parent()
+                        .unwrap()
+                        .join(import_path.string.clone())
+                };
```

Comment:
this can be simplified to:

```rs
let remapping = self
    .remappings
    .iter()
    .find(|x| import_path.string.starts_with(&x.name));

let remapped = if let Some(mappings) = remapping {
    Remapping::into_relative(mappings.clone(), &self.root)
        .path
        .relative()
        .join(import_path.string.replacen(&mappings.name, "", 1))
} else {
    current_path
        .parent()
        .unwrap()
        .join(import_path.string.clone())
};
```

Reply:
Thanks!, just applied to this and the `Import::Rename` match blockl